integration-tests/cli/discount-codes.it.js

```diff
@@ -40,77 +35,37 @@ describe('DiscountCode tests', () => {
         projectKey,
         credentials,
       }
-      // Create client and service
-      service = createRequestBuilder({ projectKey })
-      client = createClient({
-        middlewares: [
-          createAuthMiddlewareForClientCredentialsFlow(apiConfig),
-          createHttpMiddleware({ host: apiConfig.apiUrl }),
-        ],
-      })
-
+      // Clear all discount codes
       return clearData(apiConfig, 'discountCodes')
     })
+    // Clear all cart discounts
     .then(() => clearData(apiConfig, 'cartDiscounts'))
     .then(() => {
+      // Create cart-discount
       const cartDiscountDraft = fs.readFileSync(
         path.join(__dirname, './helpers/cartDiscountDraft.json'), 'utf8',
       )
-      const req = {
-        uri: service.cartDiscounts.build(),
-        method: 'POST',
-        body: cartDiscountDraft,
-      }
-      // Create cart-discount
-      return client.execute(req)
+      // Wrap in an array because the util function expects an array
+      return createData(apiConfig, 'cartDiscounts', [cartDiscountDraft])
     })
     .then((data) => {
-      cartDiscount = data.body
+      cartDiscount = data[0].body
     })
     .catch(process.stderr.write),
   )
 
-  afterAll(() => {
-    // Delete Discount codes
-    const req = {
-      uri: service.discountCodes.perPage(500).build(),
-      method: 'GET',
-    }
-    return client.execute(req)
-      .then((response) => {
-        const codes = response.body.results
-        return Promise.map(codes, (codeObj) => {
-          const deleteCodeRequest = {
-            uri: service
-              .discountCodes
-              .byId(codeObj.id)
-              .withVersion(codeObj.version)
-              .build(),
-            method: 'DELETE',
-          }
-          return client.execute(deleteCodeRequest)
-        })
-      })
-      .then(() => {
-        // Delete cart discount
-        const deleteCartRequest = {
-          uri: service
-            .cartDiscounts
-            .byId(cartDiscount.id)
-            .withVersion(1)
-            .build(),
-          method: 'DELETE',
-        }
-        return client.execute(deleteCartRequest)
-      })
-      .catch(process.stderr.write)
-  })
+  // Delete Discount codes
+  afterAll(() => clearData(apiConfig, 'discountCodes')
+    // Delete cart discounts
+    .then(() => clearData(apiConfig, 'cartDiscounts'))
+    .catch(process.stderr.write),
```

Comment:
@junajan This is after all the test cases have run. And knowing how jest handles rejected promises, not catching it here will be a big problem